client/main.c

```diff
@@ -522,50 +522,68 @@ __dump_fake_xml(const ni_dbus_variant_t *variant, unsigned int indent, const cha
 	}
 }
 
-static xml_node_t *
-__dump_object_xml(const char *object_path, const ni_dbus_variant_t *variant, ni_xs_scope_t *schema, xml_node_t *parent)
+static ni_bool_t
+__dump_object_xml(const char *object_path, const ni_dbus_variant_t *variant, 
+	ni_xs_scope_t *schema, xml_node_t *parent, const ni_string_array_t *filter)
 {
 	xml_node_t *object_node;
 	ni_dbus_dict_entry_t *entry;
 	unsigned int index;
+	const char *ifname, *interface_name;
 
 	if (!ni_dbus_variant_is_dict(variant)) {
 		ni_error("%s: dbus data is not a dict", __func__);
-		return NULL;
+		return FALSE;
 	}
 
-	object_node = xml_node_new("object", parent);
-	xml_node_add_attr(object_node, "path", object_path);
+	object_node = xml_node_new("object", NULL);
+	xml_node_add_attr(object_node, "path", object_path);	
```

Comment:
please remove the leading spaces. you may configure .gitconfig to show (and or fix) them, e.g. _core.whitespace = fix,indent-with-non-tab,trailing-space_